package-build.el

```diff
@@ -190,8 +190,9 @@ optionally looking only as far back as BOUND."
 (defun package-build--find-version-newest (regex &optional bound)
   "Find the newest version matching REGEX before point, optionally stopping at BOUND."
   (let ((tags (split-string
-                (buffer-substring-no-properties
-                  (or bound (point-min)) (point)))))
+               (buffer-substring-no-properties
+                 (or bound (point-min)) (point))
+                "\n")))
```

Comment:
I'm not sure if Git allows spaces in tags, but darcs does and the darcs repository itself has some wacky tags:
```
stable 2008-11-10
2.1.1rc2
2.1.1rc1
unstable before bytestring patches.
2.1.0
2.1.0pre3
2.1.0pre2
2.0.3pre1
this version works.
working version.
```
Changing the tag parser to split by newlines rather than by whitespace generally should prevent '2008-11-10' from being extracted as a valid tag.